test/OrchardCore.Commerce.Tests.UI/Tests/OrderTests/BehaviorOrderTests.cs

```diff
@@ -0,0 +1,157 @@
+                context
+                    .Get(By.Name("OrderPart.LineItems[0].UnitPriceCurrencyIsoCode"))
+                    .GetValue()
+                    .ShouldBe("EUR");
+
+                // For a Product with existing attributes, attributes should be populated and should be selectable.
+                await context.SetDropdownByTextAsync(By.Name("OrderPart.LineItems[0].SelectedAttributes[Size]"), "Medium");
+                await context.ClickPublishAsync();
+
+                context
+                    .Get(By.Name("OrderPart.LineItems[0].SelectedAttributes[Size]"))
+                    .GetValue()
+                    .ShouldBe("Medium");
+
+                await AddNewProductAsync(context);
+
+                await context.ClickAndFillInWithRetriesAsync(By.Name("OrderPart.LineItems[1].Quantity"), "5");
+                await context.ClickAndFillInWithRetriesAsync(By.Name("OrderPart.LineItems[1].ProductSku"), "testproduct");
+                await context.ClickAndFillInWithRetriesAsync(By.Name("OrderPart.LineItems[1].UnitPriceValue"), "10");
+                await context.ClickPublishAsync();
+
+                // No attributes should be loaded for Product without attributes.
+                context.Missing(By.Name("OrderPart.LineItems[1].SelectedAttributes[Size]"));
+
+                await context.SetDropdownByTextAsync(By.Name("OrderPart.LineItems[1].UnitPriceCurrencyIsoCode"), "CAD");
+                await context.ClickPublishAsync();
+
+                // Selecting non-matching currencies should result in validation errors.
+                VerifyValidationErrors(context, "Selected currencies need to match.");
+
+                // Links in Product Name cells should lead to content item editors.
+                context
+                    .Get(By.XPath("//a[contains(., 'Test Price Variant Product')]"))
+                    .GetAttribute("href")
+                    .ShouldContain("/Admin/Contents/ContentItems/testpricevariantproduct000/Edit");
+
+                context
+                    .Get(By.XPath("//a[contains(., 'Test Product')]"))
+                    .GetAttribute("href")
+                    .ShouldContain("/Admin/Contents/ContentItems/testproduct000/Edit");
+
+                // Product should be deletable from list after it has been submitted.
+                await context.ClickReliablyOnAsync(By.XPath("//button[contains(@class, 'btn btn-danger')][1]"));
+                await context.ClickPublishAsync();
+
+                context.Missing(By.Id("OrderPart.LineItems[1].Quantity"));
+            },
+            browser);
+
+    private static Task AddNewProductAsync(UITestContext context) => context.ClickReliablyOnAsync(By.Id("addButton"));
+
+    private static void VerifyValidationErrors(UITestContext context, string errorMessage) =>
+        context.Exists(By.XPath($"//div[contains(@class, 'validation-summary-errors')][contains(., '{errorMessage}')]"));
```

Comment:
Can you use the `context.ErrorMessageExists(errorMessage)` extension method instead?